### PR TITLE
bugfix logging event NaN error in mass balance check

### LINKED ISSUE #961: Logging event code > 10000 is not working due to mass balance error
We (Dr. Anthony Walker and I) were trying to perform logging on a specific day (icode > 10000) and we are getting a mass balance error, as follows: 

```
Logging Event Enacted on date: 12-01-1855
 mass balance error detected
 element type (see PRTGenericMod.F90):            1
 error fraction relative to biomass stock:    0.0000000000000000
 absolut error (flux in - change):                        NaN
 call index:            3
 Element index (PARTEH global):           1
 net:   -148065.34787676064
 dstock:                        NaN
 seed_in:    5.4794520547945195E-002
 net_root_uptake:   -2.7694997765899812E-016
 gpp_acc:    43.792761064199176
 flux_generic_in:    0.0000000000000000
 wood_product:    148069.15841267366
 error from patch resizing:    0.0000000000000000
 burn_flux_to_atm:    0.0000000000000000
 seed_out:    0.0000000000000000
 flux_generic_out:    0.0000000000000000
 frag_out:    14.546684805935037
 aresp_acc:    25.490334865775818
 error=net_flux-dstock:                       NaN
 biomass   2913.8406328922529
 litter                       NaN
 seeds   237.54743874994409
 total stock                       NaN
 previous total   355119.52203159698
 lat lon   35.899999999999999        275.66669999999999
 ```
 
 In the log file, it looks like the abs error is a NaN. That would likely be causing the mass balance error. 

Could you please look into this issue and help us resolve it?

Thanks,
Bharat

### PR DESCRIPTION
**Pull request by walkeranthonyp**
### Description:

Edits to `logging_litter_fluxes` that avoid a divide by zero when all of a patch is harvested and becomes a new patch. 
Edits consistent with other litter flux calculation subroutines.  

Fixes #961 

### Collaborators:

@sharma-bharat @ckoven @JoshuaRady 

### Expectation of Answer Changes:

Single event / annual logging will not fail in mass balance check. 

### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run
- [ ] If answers were expected to change, evaluation was performed and provided

### Test Results:

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:

FATES baseline hash-tag:

Test Output:

### CODE REVIEW

File: biogeochem/EDLoggingMortalityMod.F90 [hunk 1]
```diff
@@ -510,17 +516,23 @@ subroutine logging_litter_fluxes(currentSite, currentPatch, newPatch, patch_site
...
 
-
       ! Calculate the fraction of litter to be retained versus donated
       ! vis-a-vis the new and donor patch
-      
       retain_frac = (1.0_r8-harvest_litter_localization) * &
             remainder_area/(newPatch%area+remainder_area)
       donate_frac = 1.0_r8-retain_frac
 
+      if(remainder_area > rsnbl_math_prec) then
```
Comment (rgknox) [inline biogeochem/EDLoggingMortalityMod.F90:527]: we might want to have these types of checks in the fire and treefall mortality patch disturbance pipelines too

**Review by sharma-bharat — APPROVED** [review]

File: biogeochem/EDLoggingMortalityMod.F90 [hunk 2]
```diff
@@ -474,34 +478,36 @@ subroutine logging_litter_fluxes(currentSite, currentPatch, newPatch, patch_site
...
+      real(r8) :: bg_wood                ! below ground wood mass [kg]
+      real(r8) :: remainder_area         ! current patch's remaining area after donation [m2]
+      real(r8) :: leaf_m                 ! leaf element mass [kg]
+      real(r8) :: fnrt_m                 ! fineroot element mass [kg]
+      real(r8) :: sapw_m                 ! sapwood element mass [kg]
+      real(r8) :: store_m                ! storage element mass [kg]
+      real(r8) :: struct_m               ! structure element mass [kg]
+      real(r8) :: repro_m                ! reproductive mass [kg]
+      real(r8) :: retain_frac            ! fraction of litter retained in the donor patch
+      real(r8) :: donate_frac            ! fraction of litter sent to newly formed patch
```
Comment (ckoven) [inline biogeochem/EDLoggingMortalityMod.F90:500]: It looks like `donate_frac` is calculated but no longer used for anything. delete?

File: biogeochem/EDLoggingMortalityMod.F90 (hunk 1 above)
Comment (rgknox) [reply biogeochem/EDLoggingMortalityMod.F90:527]: I did a check, and we do use this logic for the fire and treefall routines.

**Review by rgknox — APPROVED** [review]

File: biogeochem/EDLoggingMortalityMod.F90 (hunk 2 above)
Comment (walkeranthonyp) [reply biogeochem/EDLoggingMortalityMod.F90:500]: Right, I think I noticed that too.

Comment (glemieux) [reply biogeochem/EDLoggingMortalityMod.F90]: removed

**Comment by glemieux** [comment]
Testing underway on cheyenne.

**Comment by glemieux** [comment]
@walkeranthonyp should https://github.com/NGEET/fates/issues/961 be closed once this pr is merged to fates `main`?

**Comment by walkeranthonyp** [comment]
Tentatively yes, the initial issue has been resolved with this PR. 

There is one thing that came up in the discussions about the effects of harvest_rate = 1 being equivalent to mort_disturb_frac = 0.

Prob best to close this and have that discussion elsewhere. 

@ckoven -- thoughts?

**Comment by glemieux** [comment]
Charlie's on vacation for the rest of the week.  I'll tag the issue to autoclose when we merge this and create a new discussion from your comments pointing back to #961.

**Comment by glemieux** [comment]
For reference, here is the discussion continuation thread: https://github.com/NGEET/fates/discussions/1003.

**Pull request merged**